code/game/objects/items/storage/storage.dm

```diff
@@ -4,43 +4,82 @@
 // Do not remove this functionality without good reason, cough reagent_containers cough.
 // -Sayu
 
-
 /obj/item/storage
 	name = "storage"
 	icon = 'icons/obj/items/storage/storage.dmi'
 	w_class = WEIGHT_CLASS_NORMAL
-	var/list/can_hold = list() //List of objects which this item can store (if set, it can't store anything else)
-	var/list/cant_hold = list() //List of objects which this item can't store (in effect only if can_hold isn't set)
-	var/list/bypass_w_limit = list() //a list of objects which this item can store despite not passing the w_class limit
-	var/list/click_border_start = list() //In slotless storage, stores areas where clicking will refer to the associated item
+	var/flags_storage = STORAGE_FLAG_FULLNESS_VISIBLE
+
+	/**
+	 * A list of all things that WILL spawn inside of this storage object on creation.
+	 * Can either be an assosciative list of items to add, or a simple list of singular items.
+	 *
+	 * Format:
+	 * 	list(
+	 *   /obj/item/XXX = 1,
+	 *   /obj/item/XXX = 2,
+	 * 	)
+	 * OR
+	 * 	list(
+	 * 	 /obj/item/XXX,
+	 *   /obj/item/XXX,
+	 * 	)
+	 *
+	 * It is generally better to use the first format if you reduce four or more lines of code doing so.
+	 */
+	var/list/spawns_with = list()
```

Comment:
youre using lazylist macros for these so you dont need them to be initialized even if not used.
```suggestion
	var/list/spawns_with
```